modules/dev-utils/src/main/java/org/apache/ignite/development/utils/IgniteWalConverter.java

```diff
@@ -94,7 +107,7 @@ public static void convert(final PrintStream out, final IgniteWalConverterArgume
 
         boolean printAlways = F.isEmpty(params.getRecordTypes());
 
-        try (WALIterator stIt = factory.iterator(iteratorParametersBuilder)) {
+        try (FilteredWalIterator stIt = walIterator(factory.iterator(iteratorParametersBuilder), params.getPages())) {
```

Comment:
We don't need to switch from abstract WALIterator interface to exact implementation. This could be hidden in `walIterator` and `getCurrentWalFilePath` implementations.

If pages were not provided by the user we don't need to use FilteredWalIterator with effectively no-op filter.